src/Symfony/Component/Validator/Util/PropertyPath.php

```diff
@@ -42,7 +42,7 @@ public static function append($basePath, $subPath)
                 return $basePath.$subPath;
             }
 
-            return $basePath ? $basePath.'.'.$subPath : $subPath;
+            return '' !== (string) $basePath ? $basePath.'.'.$subPath : $subPath;
```

Comment:
$basePath could be FALSE, or NULL also - in which case we should not add the point delimiter also.